ozaria/site/components/interactive/PageInteractive/insertCode.vue

```diff
@@ -1,34 +1,207 @@
 <script>
-export default {
-   props: {
-    interactive: {
-      type: Object,
-      required: true,
-      default: {}
+  import { codemirror } from 'vue-codemirror'
+
+  // TODO dynamically import these
+  import 'codemirror/mode/javascript/javascript'
+  import 'codemirror/mode/python/python'
+  import 'codemirror/lib/codemirror.css'
+
+  import BaseInteractiveTitle from './BaseInteractiveTitle'
+
+  const SAMPLE_CODE = `
+          let x = "y"
+
+          let a = "b"
+          console.log(x + a)
+          `
+
+  export default {
+    components: {
+      codemirror,
+      'base-interactive-title': BaseInteractiveTitle
     },
-    introLevelId: {
-      type: String,
-      required: true,
-      default: ''
+
+    props: {
+      interactive: {
+        type: Object,
+        required: true,
+        default: () => ({})
+      },
+
+      introLevelId: {
+        type: String,
+        required: true,
+        default: ''
+      },
+
+      interactiveSession: {
+        type: Object
+      },
+
+      courseInstanceId: {
+        type: String
+      }
     },
-    interactiveSession: {
-      type: Object
+
+    data () {
+      return {
+        codemirrorOptions: {
+          tabSize: 2,
+          mode: 'text/javascript', // TODO drive this from the classroom
```

Comment:
We can also pass in the language from intro level page since it is calculated there https://github.com/codecombat/codecombat/blob/master/ozaria/site/components/play/PageIntroLevel/index.vue#L72-L77